File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {   
        int[] answer = new int[2];
        
        int colorSum = brown + yellow;
        
        for (int height = 3; height < colorSum; height++) {
            if (colorSum % height == 0) {
                int width = colorSum / height;
                int expectedYellow = (width - 2) * (height - 2);
                if (expectedYellow == yellow) {
                    answer[0] = width;
                    answer[1] = height;
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int sum = brown + yellow;
    
    for (int h = 3; h <= sum; h++) {
        int w = sum / h;
        int yl = (w - 2) * (h - 2);
        int br = sum - yl;
        if (yl == yellow && br == brown) {
            answer.push_back(w);
            answer.push_back(h);
            break;
        }
    }
    return answer;
}